pgx-pg-sys/cshim/pgx-cshim.c

```diff
@@ -43,11 +43,90 @@ void pgx_elog_error(char *message) {
     elog(ERROR, "%s", message);
 }
 
-PGDLLEXPORT void pgx_ereport(int level, int code, char *message, char *file, int lineno, int colno);
-void pgx_ereport(int level, int code, char *message, char *file, int lineno, int colno) {
-    ereport(level,
-            (errcode(code),
-                    errmsg("%s", message), errcontext_msg("%s:%d:%d", file, lineno, colno)));
+PGDLLEXPORT void pgx_errcontext_msg(char *message);
+void pgx_errcontext_msg(char *message) {
+    errcontext_msg("%s", message);
+}
+
+PGDLLEXPORT void pgx_ereport(const int level, int code, char *message, char *detail, char *funcname, char *file, int lineno, char *context);
+void pgx_ereport(const int level, int code, char *message, char *detail, char *funcname, char *file, int lineno, char *context) {
```

Comment:
this is incredibly gross, I know. I was playing C preprocessor in my head to unroll the #define ereport() macro.

I think I have a cleaner approach and will take a look at that shortly.